src/Analysis/Ast/Impl/Types/PythonClassType.cs

```diff
@@ -281,5 +284,55 @@ internal PythonClassType(string name, Location location)
         public bool Equals(IPythonClassType other)
             => Name == other?.Name && DeclaringModule.Equals(other?.DeclaringModule);
 
+        private IEnumerable<IPythonType> DisambiguateBases(IEnumerable<IPythonType> bases, IExpressionEvaluator eval) {
+            if (bases == null) {
+                return Array.Empty<IPythonType>();
+            }
+
+            if (eval == null) {
+                return FilterCircularBases(bases).Where(b => !b.IsUnknown());
+            }
+
+            var newBases = new List<IPythonType>();
+            foreach (var b in bases) {
+                var imported = eval.LookupImportedNameInScopes(b.Name, out _);
+                if (imported is IPythonType importedType) {
+                    // Variable with same name as the base was imported.
+                    // If there is also a local declaration, we need to figure out which one wins.
+                    var declared = eval.LookupNameInScopes(b.Name, out var scope);
+                    if (declared != null && scope != null) {
+                        var v = scope.Variables[b.Name];
+                        if (v.Source != VariableSource.Import && v.Value is IPythonClassType cls && cls.Location.IndexSpan.Start >= Location.IndexSpan.Start) {
```

Comment:
Maybe consider an extension method for checking if a declaration/assignment occurs after a certain point, I made a similar call in `AssignmentHandler` so maybe it's getting popular enough

Reply:
Sure